olive/systems/azureml/aml_system.py

```diff
@@ -136,47 +144,59 @@ def _create_model_args(self, model_json: dict, tmp_dir: Path):
             model_json["config"]["script_dir"] = None
 
         model_path = None
-        if model_json["config"]["model_storage_kind"] == ModelStorageKind.AzureMLModel:
-            model_path = Input(
-                type=AssetTypes.CUSTOM_MODEL,
-                path=model_json["config"]["model_path"],
-            )
-            model_json["config"]["model_storage_kind"] = str(ModelStorageKind.LocalFile)
-            model_json["config"]["version"] = None
-        else:
-            if model_json["config"].get("model_path"):
-                original_model_path = Path(model_json["config"]["model_path"]).resolve()
-                if (
-                    model_json["type"].lower() == "onnxmodel"
-                    and model_json["config"]["model_storage_kind"] == ModelStorageKind.LocalFolder
-                ):
-                    # onnx model with external data
-                    # need to upload the parent directory of .onnx file
-                    original_model_path = original_model_path.parent
-                # use common name "model" for model_path
-                tmp_model_path = (tmp_dir / "model").with_suffix(original_model_path.suffix)
-                if original_model_path.is_dir():
-                    # copy model directory
-                    # symlink doesn't work for directory
-                    shutil.copytree(original_model_path, tmp_model_path, symlinks=True)
-                    if model_json["type"].lower() == "onnxmodel":
-                        # rename .onnx file to model.onnx
-                        onnx_model_file = Path(model_json["config"]["model_path"]).resolve().name
-                        (tmp_model_path / onnx_model_file).rename(tmp_model_path / "model.onnx")
-                else:
-                    # symlink model file
-                    tmp_model_path.symlink_to(original_model_path)
-                model_path = Input(
-                    type=AssetTypes.URI_FILE
-                    if model_json["config"].get("model_storage_kind") == ModelStorageKind.LocalFile
-                    else AssetTypes.URI_FOLDER,
-                    path=tmp_model_path,
+        if model_json["config"].get("model_path"):
+            model_resource_path = ResourcePath.create_resource_path(model_json["config"]["model_path"])
+            asset_type = RESOURCE_TYPE_TO_ASSET_TYPE[model_resource_path.type]
+
+            if model_resource_path.type in AZUREML_RESOURCE_TYPES:
+                # ensure that the model is in the same workspace as the system
+                model_workspace_config = model_resource_path.config.azureml_client.get_workspace_config()
+                system_workspace_config = self.azureml_client_config.get_workspace_config()
+                for key in model_workspace_config:
+                    if model_workspace_config[key] != system_workspace_config[key]:
+                        raise Exception(
```

Comment:
better not raise Exception

Reply:
What error should I raise? ValueError? We need them to be in the same workspace for this to work.

Reply:
You can find the appropriated exception at https://docs.python.org/3/library/exceptions.html#exception-hierarchy.

Some best practice is https://stackoverflow.com/questions/2052390/manually-raising-throwing-an-exception-in-python.

Reply:
Updated to `ValueError` since it depends on the input model.